src/wp-admin/includes/class-wp-screen.php

```diff
@@ -900,9 +900,70 @@ public function show_screen_options() {
 		$this->_screen_settings = '';
 
 		if ( 'post' === $this->base ) {
-			$expand                 = '<fieldset class="editor-expand hidden"><legend>' . __( 'Additional settings' ) . '</legend><label for="editor-expand-toggle">';
-			$expand                .= '<input type="checkbox" id="editor-expand-toggle"' . checked( get_user_setting( 'editor_expand', 'on' ), 'on', false ) . ' />';
-			$expand                .= __( 'Enable full-height editor and distraction-free functionality.' ) . '</label></fieldset>';
+			$expand		= '<fieldset class="editor-expand hidden"><legend>' . __( 'Additional settings' ) . '</legend><label for="editor-expand-toggle">';
+			$expand		.= '<input type="checkbox" id="editor-expand-toggle"' . checked( get_user_setting( 'editor_expand', 'on' ), 'on', false ) . ' />';
+			$expand		.= __( 'Enable full-height editor and distraction-free functionality.' ) . '</label>';
+
+			/**
+			 * Filter for Previous and Next buttons alongside Add New button.
+			 * 
+			 * @param bool Whether to show these buttons. Default true.
+			 * @param $post_type
+			 *
+			 * @since CP-1.x.x
+			 */			
+			if ( apply_filters( 'admin-post-navigation', true, $this->post_type ) ) {
+
+				$checked = 'checked="checked"';
+				$nav_display = 'inline';
+				$metaboxhidden_post = get_user_meta( get_current_user_id(), 'metaboxhidden_post', true );
+
+				if ( in_array( 'adminpostnav', $metaboxhidden_post ) ) {
+					$checked = '';
+					$nav_display = 'none';
+				}
+
+				/*
+				 * Enable overriding of default choice (made on post
+				 * screen) for other post types.
+				 * 
+				 * @param $nav_display	Must be either 'inline' or 'none'.
+				 * @param $post_type
+				 *
+				 * @since CP-1.x.x 
+				 */
+				$admin_nav = apply_filters( 'admin-post-navigation-buttons', $nav_display, $this->post_type );
+				if ( $admin_nav === 'inline' ) {
+					$checked = 'checked="checked"';
+				}
+				elseif ( $admin_nav === 'none' ) {
+					$checked = '';
+				}
+
```

Comment:
This override looks out of place here.

Reply:
It's unlike you to be so vague! What does that mean? Are you saying there shouldn't be an override? A comment in the forum expressly asked for the ability to change the setting by post type, and core doesn't permit that without an override like this.

Reply:
I was vague because I didn't run it, and I can't tell if the user choice (checkbox) is being overriden by the filter or something else. It looks odd to set the checkbox twice that way, since it looks like the user choice is ignored totally.

Reply:
The way core works is that these settings apply globally. So the buttons would be either shown or not for all post types. What this filter does is make it possible for the user to choose while also allowing for a fixed setting (which might be different) for any specific post type. Obviously the filter doesn't do anything by itself, but it does allow a developer to make a choice to have this on some post types but not on others. I can't say that I'm particularly wedded to this, but it provides the choice asked for in the forum.

Reply:
You can save one per post type if you wanted. (especially if it's not lumped together with dissimilar pre-existing code)
It seems that the Screen Option should be shown regardless, so the user can choose. I suppose this could be filtered so some post types don't give you the choice.
Then the links only have to follow the Screen Option setting (which is per post type). The markup could be filtered as mentioned in a separate comment so the order can be affected.

Reply:
Now found the correct usermeta to hook into to achieve settings on a per post type basis, so this filter is no longer necessary.